packages/cmf/src/componentState.js

```diff
@@ -0,0 +1,61 @@
+import { PropTypes } from 'react';
+import { actions } from 'react-cmf';
+
+export function getStateAccessors(dispatch, name, id, DEFAULT_STATE) {
+	return {
+		updateState(state) {
+			dispatch(
+				actions.componentsActions.mergeComponentState(
+					name,
+					id,
+					state,
+				),
+			);
+		},
+		initState(initialState) {
+			const state = DEFAULT_STATE.merge(initialState);
+			dispatch(
+				actions.componentsActions.addComponentState(
+					name,
+					id,
+					state,
+				),
+			);
+		},
+	};
+}
+
+export function getStateProps(state, name, id) {
+	return {
+		state: state.cmf.components.getIn([name, id]),
+	};
+}
+
+export function initState(props) {
+	if (!props.state && props.initState) {
+		props.initState(props.initialState);
+	}
+}
+
+// BBB
+export function stateWillMount(props) {
```

Comment:
we should make it a console.error to make the message more visible and issue a depreciation date prévision.